vault/logical_system.go

```diff
@@ -1869,6 +1899,44 @@ func (b *SystemBackend) handleAuthTable(ctx context.Context, req *logical.Reques
 	return resp, nil
 }
 
+func augmentEnableAuthConfigMap(configMap map[string]interface{}) error {
+	configParamNameSlice := []string{
+		"audit_non_hmac_request_keys",
+		"audit_non_hmac_response_keys",
+		"passthrough_request_headers",
+		"allowed_response_headers",
+	}
+	for _, paramName := range configParamNameSlice {
+		if raw, ok := configMap[paramName]; ok {
+			outputSlice := []string{}
+
+			switch raw.(type) {
+			case []interface{}:
+				rawNew := raw.([]interface{})
+				for _, rawVal := range rawNew {
+					rawValSt := rawVal.(string)
```

Comment:
Are we safe to assume that the []interface{} will always only contain string elements?  I think so, but if you're not sure, it's always better to do a conditional cast and return an error if it's not the right type.  Better than a surprising panic.

Reply:
Done